### ISSUE TITLE
Word pretrained embedding 

### ISSUE THREAD

**Issue by bxclib2** [OpenNMT/OpenNMT-py#1592]
Hello

I have a relatively small training data in my usage of openNMT.py (About 10000 sentences.)

I would like to use the pretrained embedding (Glove, Fasttext, etc).

When I do translation, there are a lot of unseen words(Since the training data is relatively small.)

However, as far as I am concerned, when doing translation, if we meet a word which is not seen in training data but it is indeed in the Glove vocabulary, giving the Glove embedding of that word is better than just give all zero embedding.

However, as your tutorial, the vocabulary is "fixed" during data preprocessing time, so the embedding vectors for those words don't appear in the training and validation set are directly removed. Could you find some way to change this feature?

I hope that I made this issue clear. Thank you for your help!

Best regards

**Comment by vince62s** [comment]
Changing vocab is not supported yet.
Anyway, with this dataset size I doubt you'll get anything good.

**Comment by sahinurlaskar** [comment]
How do i use Fasttext word embeddings in OpenNMT-py

**Comment by sahinurlaskar** [comment]
Does OpenNMT-py supports Fasttext word embeddings?

**Comment by francoishernandez** [comment]
No it does not.
Though other pretrained embeddings type are supported, as mentioned [in the docs](https://opennmt.net/OpenNMT-py/FAQ.html#how-do-i-use-pretrained-embeddings-e-g-glove). So, you could probably adapt the loading function to your use case:
https://github.com/OpenNMT/OpenNMT-py/blob/d8ddbdf02382cc0e2da1cb8e1d2762a4bc800728/onmt/modules/embeddings.py#L307
